From: stern@rowland.harvard.edu
Subject: [PATCH] USB: UHCI: Fix build errors when CONFIG_DEBUG_FS isn't set

[PATCH] USB: UHCI: Fix build errors when CONFIG_DEBUG_FS isn't set

> I've dropped this now.  It caused the usb-bk tree to be dropped from the
> -mm releases due to it causing build errors if CONFIG_DEBUG_FS not being
> enabled.  Andrew sent me a horrible hack to try to fix it up, but I'm
> going to trust you to fix it up properly :)

Here's a proper fix (well, it's a little hackish, but better than Andrew's
I hope).  No more build problems if CONFIG_DEBUG_FS isn't set.


Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/usb/host/uhci-debug.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)


diff -Nru a/drivers/usb/host/uhci-debug.c b/drivers/usb/host/uhci-debug.c
--- a/drivers/usb/host/uhci-debug.c	2005-03-08 16:51:15 -08:00
+++ b/drivers/usb/host/uhci-debug.c	2005-03-08 16:51:15 -08:00
@@ -17,6 +17,8 @@
 
 #include "uhci-hcd.h"
 
+static struct dentry *uhci_debugfs_root = NULL;
+
 /* Handle REALLY large printk's so we don't overflow buffers */
 static inline void lprintk(char *buf)
 {
@@ -497,8 +499,6 @@
 
 #define MAX_OUTPUT	(64 * 1024)
 
-static struct dentry *uhci_debugfs_root = NULL;
-
 struct uhci_debug {
 	int size;
 	char *data;
@@ -579,4 +579,9 @@
 	.read =		uhci_debug_read,
 	.release =	uhci_debug_release,
 };
+
+#else	/* CONFIG_DEBUG_FS */
+
+#define uhci_debug_operations (* (struct file_operations *) NULL)
+
 #endif
